sigtrap_posix.go

```diff
@@ -76,9 +76,17 @@ func trapSignalsPosix() {
 					caddyfileToUse = newCaddyfile
 				}
 
+				// Backup old event hooks
+				oldEventHooks := cloneEventHooks()
+
+				// Purge the old event hooks
+				purgeEventHooks()
+
 				// Kick off the restart; our work is done
 				_, err = inst.Restart(caddyfileToUse)
 				if err != nil {
+					eventHooks = oldEventHooks
```

Comment:
Actually there is race condition here, because pointer assignment isn't atomic on some platform.
But this is a rare case we should hit, we can just let it go for now. Maybe this is one of the TODO list.

Reply:
@tw4452852 do you have a source for that?

Reply:
We could use a mutex to prevent a race

Reply:
@tw4452852 I implemented sync.Mutex to protect `eventHooks` itself

Reply:
@tw4452852 I implemented `sync.Mutex` to protect `eventHooks` itself

Reply:
Okay, I have found a better solution :)